salt/metalk8s/kubeadm/init/kubelet-start/configured.sls

```diff
@@ -0,0 +1,62 @@
+{%- from "metalk8s/map.jinja" import kubelet with context %}
+
+Create kubelet service environment file:
+  file.managed:
+    - name: {{ kubelet.service.environment_file }}
+    - source: salt://metalk8s/kubeadm/init/kubelet-start/files/kubeadm.env
+    - template: jinja
+    - user: root
+    - group: root
+    - mode: 644
+    - context:
+        options: {{ kubelet.service.options }}
+
+{#- Default dataset for kubelet config #}
+{%- set dataset = salt['grains.filter_by']({
```

Comment:
Not sure why you're splitting the configuration in two parts, here and above.

If this has a reason, some changes will be required anyway: you specify the `kind` here, and the `apiVersion` above, whilst it's the combination of those two which defines the schema of the configuration/object, so splitting them doesn't make much sense.

Reply:
My idea was to be able to change "changing stuff" from the pillar.
The kind for kubelet configuration is always `KubeletConfiguration` no matter the `apiVersion` or i'm wrong ?

But yes I can add it in the default.yaml also make sense. Will do it as the result is the same

Reply:
Discussed offline :)